framework/src/utils/PerfGraphLivePrint.C

```diff
@@ -0,0 +1,309 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PerfGraphLivePrint.h"
+
+PerfGraphLivePrint::PerfGraphLivePrint(PerfGraph & perf_graph, MooseApp & app)
+  : ConsoleStreamInterface(app),
+    _perf_graph(perf_graph),
+    _execution_list(perf_graph._execution_list),
+    _done_future(perf_graph._done.get_future()),
+    _destructing(perf_graph._destructing),
+    _should_print(true),
+    _id_to_section_info(perf_graph._id_to_section_info),
+    _time_limit(perf_graph._live_print_time_limit),
+    _mem_limit(perf_graph._live_print_mem_limit),
+    _stack_level(0),
+    _last_execution_list_end(0),
+    _last_printed_increment(NULL)
+{
+}
+
+void
+PerfGraphLivePrint::printLiveMessage(PerfGraph::SectionIncrement & section_increment)
+{
```

Comment:
Should add something like:

```
mooseAssert(_id_to_section_info.count(section_increment.id), "Not found in map");
const auto & info = _id_to_section_info[section_increment.id];
```

and use it in the sections that follow. Same with in the `printStats` method